pkg/controllers/status/cluster_status_controller.go

```diff
@@ -139,7 +142,7 @@ func (c *ClusterStatusController) syncClusterStatus(cluster *clusterv1alpha1.Clu
 	readyCondition := c.clusterConditionCache.thresholdAdjustedReadyCondition(cluster, &observedReadyCondition)
 
 	// cluster is offline after retry timeout, update cluster status immediately and return.
-	if !online && readyCondition.Status != metav1.ConditionTrue {
+	if !online && !isConditionTrue(readyCondition) {
```

Comment:
```suggestion
	if !online && readyCondition.Status != metav1.ConditionFalse {
```
Does that make sense to you?
And, I don't think `isConditionTrue()` would help, can we remove it?